### PR TITLE
stage2: clean up test harness

### PR DESCRIPTION
**Pull request by pixelherodev**

### CODE REVIEW

**Comment by andrewrk** [comment]
what's going on here?

**Comment by pixelherodev** [comment]
This should be three commits, I'm going to split it up. That should make
it clearer.

-- Email domain proudly hosted at https://migadu.com

File: src-self-hosted/test.zig [hunk 1]
```diff
@@ -415,26 +413,25 @@ pub const TestContext = struct {
...
+    fn runOneCase(self: *TestContext, allocator: *Allocator, root_node: *std.Progress.Node, case: Case) !void {
         var tmp = std.testing.tmpDir(.{});
         defer tmp.cleanup();
 
         const tmp_src_path = if (case.extension == .Zig) "test_case.zig" else if (case.extension == .ZIR) "test_case.zir" else unreachable;
         const root_pkg = try Package.create(allocator, tmp.dir, ".", tmp_src_path);
         defer root_pkg.destroy();
 
-        const bin_name = try std.zig.binNameAlloc(allocator, "test_case", target, case.output_mode, null);
+        const bin_name = try std.zig.binNameAlloc(allocator, "test_case", case.target.toTarget(), case.output_mode, null);
```
Comment (andrewrk) [inline src-self-hosted/test.zig:429]:
```zig
    /// TODO deprecated, use `std.zig.system.NativeTargetInfo.detect`.
    pub fn toTarget(self: CrossTarget) Target {
        return .{
            .cpu = self.getCpu(),
            .os = self.getOs(),
            .abi = self.getAbi(),
        };
    }
```

Comment (pixelherodev) [reply src-self-hosted/test.zig]: Yeah, looking more closely at the diff for that commit made it obvious what I was doing wrong.

Good now.

**Comment by andrewrk** [comment]
I think e4eb4396c27010d8d07b6fc73e7c7557170f06ad does what you were going for here.

**Pull request closed without merging**